Task03 one piece website

EstebanMadrigalAK opened the pull request:
# Pull Request
## Description
This branch contains the code for the one piece application with mobile, tablet and desktop mode, there is a style folder where you can find the style files, also there is the html and js files corresponding to the app

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?
I test the functionality of the application with help of an extension called live server that helps to see the changes of the html and how it looks.

- install the extension
- open the html file you want to test on vsc
- select the go live option on the bottom panel
- it will open the web browser showing the html
- every change on the html or css is going to be showed on the browser window

EstebanMadrigalAK committed:
New images added

EstebanMadrigalAK committed:
Merge branch 'Task02-PixelPerfect' into Task03-OnePieceWebsite

EstebanMadrigalAK committed:
main files for task 3

EstebanMadrigalAK committed:
Navbar done

Adding images for the navbar and header

EstebanMadrigalAK committed:
restructure styles and header done

Creating scss for each component and header done

EstebanMadrigalAK committed:
sipnosis section implemented, no done yet

EstebanMadrigalAK committed:
Sipnosis with read more done

EstebanMadrigalAK committed:
Logo on read more added

EstebanMadrigalAK committed:
little bug fixed

EstebanMadrigalAK committed:
Searchbar implemented

EstebanMadrigalAK committed:
Merge branch 'Task02-PixelPerfect' into Task03-OnePieceWebsite

EstebanMadrigalAK committed:
removing decimals

EstebanMadrigalAK committed:
HTML and sizes fixed

EstebanMadrigalAK committed:
Sinopsis section fixed

EstebanMadrigalAK committed:
Searchbar done

EstebanMadrigalAK committed:
characters section done

EstebanMadrigalAK committed:
characters and island sections done

EstebanMadrigalAK committed:
misto-objects done

EstebanMadrigalAK committed:
footer done

EstebanMadrigalAK committed:
move content style

EstebanMadrigalAK committed:
navbar code refactored

EstebanMadrigalAK committed:
Search bar refactored

EstebanMadrigalAK committed:
mist object refactored

EstebanMadrigalAK committed:
footer refactored

EstebanMadrigalAK committed:
text-properties mixin

EstebanMadrigalAK committed:
flex-column-centered mixin

EstebanMadrigalAK committed:
sass restructured

EstebanMadrigalAK committed:
islands more responsive on tablet

EstebanMadrigalAK committed:
Searchbar and sipnosis scss files created

EstebanMadrigalAK committed:
Mobile done

EstebanMadrigalAK committed:
Starting using padding and margin

EstebanMadrigalAK committed:
tablet mode done

EstebanMadrigalAK committed:
footer fixed

EstebanMadrigalAK committed:
Desktop mode done

EstebanMadrigalAK committed:
Burger menu navbar

EstebanMadrigalAK committed:
navbar for mobile done

EstebanMadrigalAK committed:
navbar on tablet mode

EstebanMadrigalAK committed:
navbar for desktop working

EstebanMadrigalAK committed:
refactoring code

EstebanMadrigalAK committed:
removing spaces

EstebanMadrigalAK committed:
logo on read all done

EstebanMadrigalAK committed:
adding link to the other pages

falvarado3 commented on Task03-OnePieceWebsite/home-page.html:
```diff
@@ -0,0 +1,217 @@
+            <div class="lang-option-container">
+                <button class="lang-option">
+                    <small class="navbar-option" id="lang-text">English</small>
+                    <img src="./svg/usd.svg" alt="usd">
```
Let's try to be more specific with alt texts, these are important for accessibility in the future, I will use something like "United States flag icon"

falvarado3 commented on Task03-OnePieceWebsite/home-page.html:
```diff
@@ -0,0 +1,217 @@
+                </button>
+                <button class="lang-option">
+                    <small class="navbar-option" id="lang-text">Spanish</small>
+                    <img src="./svg/spn.svg" alt="spn">
```
The same for this and the following alts

falvarado3 commented on Task03-OnePieceWebsite/home-page.js:
```diff
@@ -0,0 +1,43 @@
+ */
+ function showSipnosis() {
+    let moreText = document.getElementById("extra-sipnosis-description");
+    let btnText = document.getElementsByClassName("read-all-label")[0];
+    let icon = document.getElementsByClassName("material-icons")[0];
+    let logo = document.getElementsByClassName("logo-header")[0];
```
I will use
`document.querySelector('.reall-all-label')` for example, this will get the first match on the document, the same for the following

EstebanMadrigalAK replied:
Should I also replace the .getElementById with the .querySelector?

falvarado3 replied:
If you want to, the change was for getting elements by class name, because it will return an array and then you have to get the first element, so the query selector is more direct because it will give you the first match. querySelectorAll will give all coincidences into an array just in case.
So for an id, it's your choice if you want to use the query or id selector. 😁

EstebanMadrigalAK replied:
Ok ok thanks!

falvarado3 commented on Task03-OnePieceWebsite/home-page.js:
```diff
@@ -0,0 +1,43 @@
+ * @summary short description for the file
+ */
+ function showSipnosis() {
+    let moreText = document.getElementById("extra-sipnosis-description");
```
Use const instead, if the value is never reassigned, the same for all values if rule applies

falvarado3 commented on Task03-OnePieceWebsite/style/layouts/_content.scss:
```diff
@@ -0,0 +1,119 @@
+@import '../components/searchbar';
+@import '../sections/sipnosis';
+
+.content{
```
remember leaving a space before '{', same for all cases in following files

EstebanMadrigalAK replied:
On the last PR review you mention the spaces after the brackets and i thought you meant to remove them so i remove all of them for that, im adding them right now

falvarado3 replied:
Sorry! maybe I confused after with before haha, but yeah add a space before the bracket
This is tedious I know, on future projects some tools will take care of this for you! 🥳

EstebanMadrigalAK replied:
Haha ok ok no worries, we are cleared now

falvarado3 reviewed (commented):
Really nice implementation! Like the idea of separated scss files. Looks clean and organized.
Take a look at the few comments I left.

EstebanMadrigalAK committed:
Following feedback from PR reviewer comments

Adding space after brackets
More significant alts
Replace let for const where correspond
Replace getByClassname for querySelector
